cmd/metal-api/internal/service/machine-service.go

```diff
@@ -2316,13 +2390,19 @@ func machineHasIssues(m *v1.MachineResponse) bool {
 	return false
 }
 
-func makeMachineResponse(m *metal.Machine, ds *datastore.RethinkStore, logger *zap.SugaredLogger) *v1.MachineResponse {
-	s, p, i, ec := findMachineReferencedEntities(m, ds, logger)
-	return v1.NewMachineResponse(m, s, p, i, ec)
+func makeMachineResponse(m *metal.Machine, ds *datastore.RethinkStore) (*v1.MachineResponse, error) {
+	s, p, i, ec, err := findMachineReferencedEntities(m, ds)
+	if err != nil {
+		return nil, err
+	}
+	return v1.NewMachineResponse(m, s, p, i, ec), nil
```

Comment:
Check if v1.NewMachineResponse should return an Error as well, or if findMachineReferencedEntities ensures to return either error or none of the return values are non nil